### ISSUE TITLE
Add support for GOG version

### ISSUE THREAD

**Issue by themrdemonized** [herbatka/HerbatasDLCModLoader#2]
The mod does nothing on GOG version, i guess since the tool generates Steam specific files. I tried to add dlc into GOG specific config but it didnt work.

I uploaded the files for the GOG loader and my mod, see if they can be of any help. Also i uploaded my mod i used for tests. Maybe the problem is in the mod itself, I can't be sure of that. Thanks in advance.

[GOGDLCLoad.zip](https://github.com/user-attachments/files/18110572/GOGDLCLoad.zip)
[zCheaperUpgrades.zip](https://github.com/user-attachments/files/18110573/zCheaperUpgrades.zip)

**Comment by herbatka** [comment]
Hey, did you try the beta version from the dev branch?
https://github.com/herbatka/HerbatasDLCModLoader/releases/tag/dev-1.0.3

Also, please provide VirusTotal scans for the files you attached.

**Comment by themrdemonized** [comment]
The dev version didnt work either. The pak it generated looks similar to my edits, just the id of the dlc mod was different

https://www.virustotal.com/gui/file/ead5282611b74d407ad387c26d6ae05bd2b7ff7cad136e0d143517f5122b54df
https://www.virustotal.com/gui/file/9c447f53e15e890d3f1b1e45f97d1491f0b510ee301b9ab15fc918de8ba05a92

**Comment by herbatka** [comment]
Alright, I'll take a closer look when I get off work, I'll get back to you in a couple hours

**Comment by herbatka** [comment]
@themrdemonized  In your .ini file, replace the 480 with 1529799785 and tell me if it worked.

**Comment by herbatka** [comment]
or just try it with this release:
https://github.com/herbatka/HerbatasDLCModLoader/releases/tag/dev-1.0.3-1

**Comment by themrdemonized** [comment]
Nope, that didnt work.
Is the mod itself fine?

**Comment by herbatka** [comment]
Yeah, loaded fine for me, can you show your folder structure? ie. How you're installing the mod?
And also send me the .pak file that the tool generates.

**Comment by themrdemonized** [comment]
![изображение](https://github.com/user-attachments/assets/00b3a187-df22-466d-bb2c-f544d0d3c5d7)
![изображение](https://github.com/user-attachments/assets/80c4e5a1-e4bf-4a8c-854c-a06e882810e0)
[HerbatasDLCModLoader.zip](https://github.com/user-attachments/files/18117054/HerbatasDLCModLoader.zip)

https://www.virustotal.com/gui/file-analysis/ZThhZjNiNjc2YzE0NWNhZGY2YmRhY2FmNWUwMTEzN2M6MTczNDAzNDg0Mg==

**Comment by herbatka** [comment]
Hmm, this is weird.
Could you please provide me with your original GOGPlatformProviderEngine.ini file?
It is located in Stalker2\Plugins\PlatformProviderController\Config\GOG inside the pakchunk0-Windows.pak file in your game Paks, I suppose you know how to unpack that?

**Comment by themrdemonized** [comment]
[GOGPlatformProviderEngine.zip](https://github.com/user-attachments/files/18123813/GOGPlatformProviderEngine.zip)

https://www.virustotal.com/gui/file-analysis/NjA3MmMwZWE2NDkwYjkyN2U5M2RmYjhhMmI4YjRjYTA6MTczNDA4MDQ0Ng==

**Comment by amcghee1** [comment]
Do you think GOG'ers are screwed herbatka, is there anything anyone can do to assist?

**Comment by amcghee1** [comment]
I've just managed to get this working using this code:
1983524879

Now, this is the bonus OST & Artbook 'DLC' product ID below so I don't know if this will work for people with editions that don't have this:
https://www.gogdb.org/product/1529799785/build/58206918472375064

This appears to be included in Deluxe & Ultimate only:
https://www.stalker2.com/editions/digital

**Comment by herbatka** [comment]
> I've just managed to get this working using this code: 1983524879
> 
> Now, this is the bonus OST & Artbook 'DLC' product ID below so I don't know if this will work for people with editions that don't have this: https://www.gogdb.org/product/1529799785/build/58206918472375064
> 
> This appears to be included in Deluxe & Ultimate only: https://www.stalker2.com/editions/digital

I'll make another release of the tool for people that own those.

Do you by chance own any DLC for other games on GOG and could check if their ID's work?

**Comment by amcghee1** [comment]
Thanks a lot. Hopefully if there's no other workaround, if GSC releases some free DLC the ID for that may work for everyone.

If you mean stalker specifically I only have S2 on GOG but do have all the current DLC for it. The rest of the DLC though are worse candidates as I see it. You have pre-order which seems to no longer be available, and the ultimate edition which already has the soundtrack part. But I'm happy to try any other IDs you think may work.

You may want to think about a mod load order function in this mod since something like Vortex doesn't see DLC loaded mods. I see for example your HCWU & OXA conflict even as DLC mods. I moved OXA after HCWU within the GOG file to get the functionality of that mod back.

**Comment by herbatka** [comment]
I meant for you to check if you could use a DLC ID from another game's DLC on GOG, if you own one that is. Like for example if you own The Witcher 3: Wild Hunt - Hearts of Stone  check if you can use 1441355562.

Also If you own the Ultimate Edition, please check if this would work: 1891718289
It's the ID for the Deluxe DLC, which is also included if you own Ultimate.

**Comment by amcghee1** [comment]
This works:
1891718289

These IDs from other games don't work (I own the bottom one only):
1441355562
1408237434
